### PR TITLE
PIM-7824: Fix the option filter

### PR DESCRIPTION
**Pull request by Doodoune**
**Description**

When an attribute option, used in the filter of a view, is removed => the view cannot be loaded.

**Definition Of Done**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | OK
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | OK
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/FilterBundle/Filter/ProductValue/ChoiceFilter.php [hunk 1]
```diff
@@ -63,11 +70,32 @@ public function apply(FilterDatasourceAdapterInterface $ds, $data)
...
+                    return $attributeOptions['code'];
+                },
+                $attributeOptions
+            );
+
+            $filteredValues = array_values(array_intersect($data['value'], $optionCodes));
+        } else {
+            $filteredValues = $data['value'];
+        }
+
```
Comment (pierallard) [inline src/Pim/Bundle/FilterBundle/Filter/ProductValue/ChoiceFilter.php]: can you put all this logic into a private method explaining what it does? For the future developers... Should be easier to read.

File: src/Pim/Bundle/FilterBundle/Filter/ProductValue/ChoiceFilter.php [hunk 2]
```diff
@@ -63,11 +70,32 @@ public function apply(FilterDatasourceAdapterInterface $ds, $data)
...
+        /* TODO @merge remove null condition on master*/
+        if (null !== $this->attributeOptionRepository &&
+            (Operators::IN_LIST === $operator || Operators::NOT_IN_LIST === $operator)
+        ) {
+            $attribute = $this->getAttribute();
+            $attributeOptions = $this->attributeOptionRepository->findCodesByIdentifiers(
+                $attribute->getCode(),
+                $data['value']
+            );
+            $optionCodes = array_map(
```
Comment (AymericPlanche) [inline src/Pim/Bundle/FilterBundle/Filter/ProductValue/ChoiceFilter.php]:
```suggestion
            $optionCodes = array_column(attributeOptions, 'code');
```

Comment (Doodoune) [reply src/Pim/Bundle/FilterBundle/Filter/ProductValue/ChoiceFilter.php]: Fix with `$attributeOptions` ;)

File: src/Pim/Bundle/FilterBundle/Filter/ProductValue/ChoiceFilter.php (hunk 1 above)
Comment (Doodoune) [reply src/Pim/Bundle/FilterBundle/Filter/ProductValue/ChoiceFilter.php]: I've added a new private method, can you take a look?

**Pull request merged**